api/v1alpha1/bmcsettings_types.go

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BMCSettings refer's to Out-Of-Band_management like IDrac for Dell, iLo for HPE etc Settings
+
+// BMCSettingsSpec defines the desired state of BMCSettings.
+type BMCSettingsSpec struct {
+	// Version contains BMC version this settings applies to
+	// +required
+	Version string `json:"version"`
+
+	// SettingsMap contains bmc settings as map
+	// +optional
+	SettingsMap map[string]string `json:"settings,omitempty"`
+
+	// BMCRef is a reference to a specific BMC to apply setting to.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="serverRef is immutable"
+	BMCRef *corev1.LocalObjectReference `json:"BMCRef,omitempty"`
+
+	// ServerMaintenancePolicy is maintenance policy to be enforced on the server when applying setting.
+	// ServerMaintenancePolicyOwnerApproval is asking for User approval for changing BMC settings
+	//	note: User approval is only enforced for server's which are reserved state
+	// ServerMaintenancePolicyEnforced will will bypass user approval and apply setting directly
+	ServerMaintenancePolicy ServerMaintenancePolicy `json:"serverMaintenancePolicy,omitempty"`
+
+	// ServerMaintenanceRefList are references to a ServerMaintenance objects that Controller has requested for the each of the related server.
+	ServerMaintenanceRefList []ServerMaintenanceRefItem `json:"serverMaintenanceRefList,omitempty"`
```

Comment:
This should be called `ServerMaintenanceRefs`.

Reply:
/ref https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions

> The name of a field referring to another resource of kind Foo by name should be called fooName. The name of a field referring to another resource of kind Foo by ObjectReference (or subset thereof) should be called fooRef.